### PR TITLE
Backport of website_portal_sale from v9

### PR DESCRIPTION
**Pull request by LeartS**
To be rebased on top of https://github.com/OCA/website/pull/68

### CODE REVIEW

**Comment by pedrobaeza** [comment]
We have talked about putting this on ecommerce repo, doesn't it?

**Comment by LeartS** [comment]
Actually this is a frontend portal for sales and invoices that can be installed and used without having the ecommerce installed

**Comment by pedrobaeza** [comment]
OK then,

Can you please check Travis?

**Comment by LeartS** [comment]
Both Travis and Runbot fail because this PR (and module) depend on https://github.com/OCA/website/pull/68 . We should merge it and then retry the build.

**Comment by pedrobaeza** [comment]
Please rebase

**Comment by rafaelbn** [comment]
@LeartS could you rebase ? https://github.com/OCA/website/pull/68 is merged. Thanks

**Comment by rafaelbn** [comment]
@LeartS runbot fails

```
2015-10-14 19:16:19     CRITICAL    server  openerp.service.server:942 preload_registries

Failed to initialize database `3117453-69-6b42d1-all`.
Traceback (most recent call last):
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3117453-69-6b42d1/openerp/service/server.py", line 929, in preload_registries
    registry = RegistryManager.new(dbname, update_module=update_module)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3117453-69-6b42d1/openerp/modules/registry.py", line 370, in new
    openerp.modules.load_modules(registry._db, force_demo, status, update_module)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3117453-69-6b42d1/openerp/modules/loading.py", line 355, in load_modules
    loaded_modules, update_module)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3117453-69-6b42d1/openerp/modules/loading.py", line 255, in load_marked_modules
    loaded, processed = load_module_graph(cr, graph, progressdict, report=report, skip_modules=loaded_modules, perform_checks=perform_checks)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3117453-69-6b42d1/openerp/modules/loading.py", line 179, in load_module_graph
    _load_data(cr, module_name, idref, mode, kind='demo')
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3117453-69-6b42d1/openerp/modules/loading.py", line 118, in _load_data
    tools.convert_file(cr, module_name, filename, idref, mode, noupdate, kind, report)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3117453-69-6b42d1/openerp/tools/convert.py", line 901, in convert_file
    convert_xml_import(cr, module, fp, idref, mode, noupdate, report)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3117453-69-6b42d1/openerp/tools/convert.py", line 987, in convert_xml_import
    obj.parse(doc.getroot(), mode=mode)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3117453-69-6b42d1/openerp/tools/convert.py", line 853, in parse
    self._tags[rec.tag](self.cr, rec, n, mode=mode)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3117453-69-6b42d1/openerp/tools/convert.py", line 755, in _tag_record
    f_val = self.id_get(cr, f_ref)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3117453-69-6b42d1/openerp/tools/convert.py", line 832, in id_get
    res = self.model_id_get(cr, id_str, raise_if_not_found)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3117453-69-6b42d1/openerp/tools/convert.py", line 843, in model_id_get
    raise_if_not_found=raise_if_not_found)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3117453-69-6b42d1/openerp/api.py", line 250, in wrapper
    return old_api(self, *args, **kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3117453-69-6b42d1/openerp/addons/base/ir/ir_model.py", line 911, in xmlid_to_res_model_res_id
    return self.xmlid_lookup(cr, uid, xmlid)[1:3]
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3117453-69-6b42d1/openerp/api.py", line 250, in wrapper
    return old_api(self, *args, **kwargs)
  File "<string>", line 2, in xmlid_lookup
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3117453-69-6b42d1/openerp/tools/cache.py", line 74, in lookup
    value = d[key] = self.method(*args, **kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3117453-69-6b42d1/openerp/addons/base/ir/ir_model.py", line 901, in xmlid_lookup
    raise ValueError('External ID not found in the system: %s' % (xmlid))
ParseError: "External ID not found in the system: portal.partner_demo_portal" while parsing /srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3117453-69-6b42d1/openerp/addons/website_portal_sale/data/demo.xml:5, near
<record id="sale_order_1" model="sale.order">
            <field name="partner_id" ref="portal.partner_demo_portal"/>
            <field name="partner_invoice_id" ref="portal.partner_demo_portal"/>
            <field name="partner_shipping_id" ref="portal.partner_demo_portal"/>
            <field name="user_id" ref="base.user_root"/>
            <field name="pricelist_id" ref="product.list0"/>
            <field name="state">sent</field>
            <field name="team_id" ref="sales_team.team_sales_department"/>
            <field name="date_order" eval="(DateTime.today() - relativedelta(months=1)).strftime('%Y-%m-%d %H:%M')"/>
            <field name="message_follower_ids" eval="[(4, ref('portal.partner_demo_portal'))]"/>
        </record>
```

**Comment by LeartS** [comment]
Yes, it's a problem with the demo data that references other non-existent demo data in v8. I'll fix it as soon as I have some time.

**Comment by LeartS** [comment]
I just removed the demo data. I know it's not the ideal solution, but I don't really have the time right now to check what kind of depending demo records are needed and reimplement them. I don't think the demo data for this module was that useful to begin with, honestly. There aren't any tests using it anyway.

**Comment by pedrobaeza** [comment]
:+1:

**Comment by rafaelbn** [comment]
cc @sergio-incaser @carlos-incaser @Yajo

**Comment by yajo** [comment]
Could you please check Travis?

File: website_portal_sale/README.rst [hunk 1]
```diff
@@ -0,0 +1,45 @@
+========================
+Website Portal for Sales
+========================
+
+Backport (with adjustments) of the ``website_portal_sale`` module of Odoo 9.
+
+This module add the user's sales documents in the frontend portal.
+Customers will be able to see the list and state of their quotations, sales
+and invoices.
+
```
Comment (yajo) [inline website_portal_sale/README.rst:10]: Instructions should say that you need to activate the portal and invite a customer to it, and how to do it.

**Comment by yajo** [comment]
I tested it in runbot. It's a bit confusing. To install this module, you need the old-fashioned `portal_sale`. When a portal user logs in, gets redirected to the old portal. To get to the new one, he must go to website and select his name in the top right corner and go to _my account_.

This does not seem much easy...

**Comment by rafaelbn** [comment]
@Yajo as @LeartS says

> I don't really have the time right now

Please @LeartS @Yajo talk about improvements and if agree, @yajo could PR to @LeartS to improve it.

Thanks

**Comment by LeartS** [comment]
@Yajo you don't need portal_sale, it's just that portal_sale automatically installs itself when the modules `sale`, `portal` and `payment` are installed. `portal` itself get automatically installed when `auth_signup` is installed.

Travis error seems unrelated:

```
2015-10-27 15:33:15,188 22816 ERROR openerp_test openerp.addons.base.ir.ir_ui_view: Element '<label class="control-label" for="vat" data-oe-id="960" data-oe-source-id="878" data-oe-xpath="/data/label" data-oe-model="ir.ui.view" data-oe-field="arch">' cannot be located in parent view

Error context:

View `saas_users`

[view_id: 960, xml_id: website_sale_vat_required.saas_users, model: n/a, parent_id: 878]
```

Regarding the redirect to the internal portal when logging in, I just didn't make any modification to the standard behaviour, what should be the right action when logging in? Redirect to website home, shop home, website portal?

**Comment by pedrobaeza** [comment]
Please rebase your branch to get the base build green

**Comment by rafaelbn** [comment]
I cannot test id because this: https://github.com/OCA/website/issues/98

I make:
- Open runbot
- Go to shop without login (I want to buy, to become a customer and check

```
Customers will be able to see the list and state of their quotations, sales
+and invoices.
```
- Buy as customer "Pepe", blocked in https://github.com/OCA/website/issues/98
- Go to same runbot in other sesion as admin and I uninstall module website_sale_vat_required to continue testing
- Without login as customer "Pepe" I finish buying
- As customer "Pepe" I try to loging and Odoo says "Wrong login/password" **There is no option to register**.

After this I have gone again to runbot selectiing base database **without modules** (this must be said in README in section "Configuration")
- I install website_portal_sale
- After installing I create a customer Pepe with email pepe@pepe.com
- I create a quotation to Pepe
- How must I continue? I think this are the steps must be writteng in section "Usage" in README

I thinks it's confusing if module is not explained in README

Thanks

**Comment by rafaelbn** [comment]
- Go to runbot selecting base database without modules http://3118388-69-57675b.runbot2.odoo-community.org
- As admin
- Install website_portal_sale
- After installing I create a customer Pepe with email pepe@pepe.com
- I create a quotation to Pepe
- Go To Settings - Users and create a user with email pepe@pepe.com that is not employee, without - any permission, only checkbox "portal"
- Assign a password to user pepe@pepe.com manualy
- Go in other browser to same runbot http://3118388-69-57675b.runbot2.odoo-community.org
- Without login, go to login and login with user a pass previously created
- You get:

```
Internal Server Error

The server encountered an internal error and was unable to complete your request. Either the server is overloaded or there is an error in the application.
```

**Comment by rafaelbn** [comment]
Please @StefanRijnhart  any review for this backport? Could be very useful. Thanks

**Comment by yajo** [comment]
> what should be the right action when logging in? Redirect to website home, shop home, website portal?

@LeartS I think the website portal should be OK. That way the user logs in and directly views his billing info and such.

> Please @LeartS @Yajo talk about improvements and if agree, @yajo could PR to @LeartS to improve it.

@rafaelbn @LeartS See https://github.com/LeartS/website/pull/1.

**Comment by rafaelbn** [comment]
Thanks.
Travis and runbot fails
What about https://github.com/OCA/website/pull/69#issuecomment-151805375 and https://github.com/OCA/website/pull/69#issuecomment-151808588 ?

**Comment by pedrobaeza** [comment]
Any news on this?

I think this should go on e-commerce repo according last approach.

File: website_portal_sale/__openerp__.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    'category': 'Website',
+    'summary': (
+        'Add your sales document in the frontend portal (sales order'
+        ', quotations, invoices)'
+    ),
+    'version': '1.0',
+    'author': 'Odoo SA, MONK Software',
+    'website': 'https://www.odoo.com/, http://www.wearemonk.com',
+    'depends': [
+        'sale',
```
Comment (sergio-teruel) [inline website_portal_sale/__openerp__.py:12]: Add portal dependency to fix runbot error
cc @rafaelbn

File: website_portal_sale/__openerp__.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
+{
+    'name': 'Website Portal for Sales',
+    'category': 'Website',
+    'summary': (
+        'Add your sales document in the frontend portal (sales order'
+        ', quotations, invoices)'
+    ),
+    'version': '1.0',
```
Comment (sergio-teruel) [inline website_portal_sale/__openerp__.py:8]: Change version to 8.0.1.0.0

File: website_portal_sale/static/src/less/website_portal_sale.less [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+
+.orders_vertical_align {
+    display: flex;
+    align-items: center;
+}
+
+
+.orders_label_text_align {
+    vertical-align: 15%;
+}
```
Comment (sergio-teruel) [inline website_portal_sale/static/src/less/website_portal_sale.less:11]: Add new line

**Comment by rafaelbn** [comment]
@LeartS could you close this PR and we will finish it?

**Comment by rafaelbn** [comment]
@LeartS as you don't have time can I close this PR and we will Open a new one?

File: website_portal_sale/__openerp__.py (hunk 2 above)
Comment (oscarolar) [inline website_portal_sale/__openerp__.py:12]: +1 @pedrobaeza

Comment (oscarolar) [inline website_portal_sale/__openerp__.py:12]: @nhomar

**Comment by rafaelbn** [comment]
*This PR will continue in https://github.com/OCA/website/pull/111
- https://github.com/OCA/website/pull/111#issuecomment-156539407
  > Regarding closing this PR, no problems

**Pull request closed without merging**